src/core/Slider/Slider.tsx

```diff
@@ -0,0 +1,644 @@
+    newValues.sort((a, b) => a - b);
+    setCurrentValues(newValues);
+  }, [values]);
+
+  const segments = React.useMemo(
+    () => generateSegments(currentValues, min, max),
+    [currentValues, min, max],
+  );
+
+  return (
+    <>
+      {'none' !== trackDisplayMode &&
+        segments.map((segment, index) => {
+          const leftPercent = (100.0 * (segment.left - min)) / (max - min);
+          let rightPercent = (100.0 * (segment.right - min)) / (max - min);
+          rightPercent = 100.0 - rightPercent;
+          if ('odd-segments' === trackDisplayMode && 0 !== (index + 1) % 2) {
+            return (
+              <TrackSegment
+                key={index}
+                leftPercent={leftPercent}
+                rightPercent={rightPercent}
+              />
+            );
+          } else if (
+            'even-segments' === trackDisplayMode &&
+            0 === (index + 1) % 2
+          ) {
+            return (
+              <TrackSegment
+                key={index}
+                leftPercent={leftPercent}
+                rightPercent={rightPercent}
+              />
+            );
+          }
+          return null;
+        })}
+    </>
+  );
+};
+
+/** Thumb is a local component used to show and modify the values maintained by the [[Slider]].
+ * Only one Thumb can be active at a time. A Thumb is made active when the user selects
+ * it with pointer. Whenever a Thumb is active, focused, or hovered its tooltip is shown. */
+const Thumb = ({ index, isActive }: { index: number; isActive: boolean }) => {
+  const {
+    hideTooltip,
+    min,
+    max,
+    setThumbMoveIndex,
+    toolTipFunction,
+    step,
+    values,
+    setThumbValue,
+    thumbMode,
+    toolTipPosition,
+  } = useSliderContext();
+  const thumbRef = React.useRef<HTMLDivElement>(null);
+  const handleOnKeyDown = React.useCallback(
+    (event: React.KeyboardEvent<HTMLDivElement>) => {
+      const [minVal, maxVal] = getAllowableThumbRange(
+        thumbMode,
+        values,
+        index,
+        min,
+        max,
+        step,
+      );
+
+      if (event.key === 'ArrowLeft') {
+        const newValue = Math.max(values[index] - step, minVal);
+        setThumbValue(index, newValue);
+      } else if (event.key === 'ArrowRight') {
+        const newValue = Math.min(values[index] + step, maxVal);
+        setThumbValue(index, newValue);
+      } else if (event.key === 'Home') {
+        setThumbValue(index, minVal);
+      } else if (event.key === 'End') {
+        setThumbValue(index, maxVal);
+      }
+    },
+    [thumbMode, values, index, min, max, step, setThumbValue],
+  );
+
+  const handlePointerDownOnThumb = React.useCallback(
+    (event: React.PointerEvent) => {
+      thumbRef.current?.focus();
+      setThumbMoveIndex(index);
+      event.preventDefault();
+      event.stopPropagation();
+    },
+    [index, setThumbMoveIndex],
+  );
+
+  const [hasFocus, setHasFocus] = React.useState(false);
```

Comment:
I wonder if there is a point having separate variables for that.. They both are only for showing/not showing tooltip.

Reply:
Seems easier to understand when looking at code - if you just had one value then when you lost focus or hover is lost  you would have to check the other condition using native element.
 hasFocus || isHovered
  true            false            -> show
  false           true             -> show
   false          false            -> don't show